templates/plugin/python/module.conf.erb

```diff
@@ -11,7 +11,11 @@
     <%= key %> <%= k %> <% if !!v == v %><%= v %><% else %>"<%= Shellwords.split(v).join('" "') %>"<% end %>
   <%- end -%>
   <%- else -%>
-    <%= key %> <% if !!value == value %><%= value %><% else %>"<%= Shellwords.split(value).join('" "') %>"<% end %>
+  <%- if not !!value == value and value.include?('"') -%>
```

Comment:
If I set the test case back to `    <%= key %> <% if !!value == value %><%= value %><% else %>"<%= Shellwords.split(value).join('" "') %>"<% end %>
`

I get a funky output / test failure.

```
Failures:

  1) collectd::plugin::rabbitmq package ensure :ensure => present Realm set to "RabbitMQ Management"
     Failure/Error: should contain_concat_fragment('collectd_plugin_python_conf_collectd_rabbitmq.collectd_plugin').with_content(/Realm "RabbitMQ Management"/)

       expected that the catalogue would contain Concat_fragment[collectd_plugin_python_conf_collectd_rabbitmq.collectd_plugin] with content set to /Realm "RabbitMQ Management"/ but it is set to "  Import \"collectd_rabbitmq.collectd_plugin\"\n\n  <Module \"collectd_rabbitmq.collectd_plugin\">\n    Host \"testhost.example.com\"\n    Password \"guest\"\n    Port \"15672\"\n    Realm \"RabbitMQ\" \"Management\"\n    Scheme \"http\"\n    Username \"guest\"\n  </Module>\n"
       Diff:
       @@ -1,2 +1,11 @@
       -(?-mix:Realm "RabbitMQ Management")
       +  Import "collectd_rabbitmq.collectd_plugin"
       +
       +  <Module "collectd_rabbitmq.collectd_plugin">
       +    Host "testhost.example.com"
       +    Password "guest"
       +    Port "15672"
       +    Realm "RabbitMQ" "Management"
       +    Scheme "http"
       +    Username "guest"
       +  </Module>

     # ./spec/classes/collectd_plugin_rabbitmq_spec.rb:56:in `block (4 levels) in <top (required)>'
```

At least in my use-case previously I was getting the ShellWord exception on render. It also addresses the spam issue:

Force the `!!value == value` to fail by changing to `  <%- if not value == value and value.include?('XXXXXX') -%>`

```
Failures:

  1) collectd::plugin::python::module spam module includes spam module configuration
     Failure/Error:
       should contain_concat__fragment('collectd_plugin_python_conf_spam')
         .with(content: /spam "wonderful" "lovely"/,)

       expected that the catalogue would contain Concat::Fragment[collectd_plugin_python_conf_spam] with content set to /spam "wonderful" "lovely"/ but it is set to "  Import \"spam\"\n\n  <Module \"spam\">\n    spam \"\"wonderful\" \"lovely\"\"\n  </Module>\n"
       Diff:
       @@ -1,2 +1,6 @@
       -(?-mix:spam "wonderful" "lovely")
       +  Import "spam"
       +
       +  <Module "spam">
       +    spam ""wonderful" "lovely""
       +  </Module>

     # ./spec/defines/collectd_plugin_python_module_spec.rb:77:in `block (3 levels) in <top (required)>'
```

So I modified the test case to look for a value which is a string of quoted words and ensure the output is still valid.

I'm not convinced this will solve the nested data structure stuff - got an example?